### PR TITLE
Move conversationMedia out of the contacts JSON blob into it's own table

### PR DESCRIPTION
**Pull request by GPaoloni**
## Description
This PR:
- Adds a migration to create `ConversationMedias` table.
- Refactors contact logic to create new entries on the above table if there are conversation media entries in the payload.
- Refactors all queries to include conversation media relations where a contact is retrieved.
- Refactors the contact jobs system to contemplate the above changes.
- Added migration to convert existing information to the new table - the legacy media is not deleted.

### Checklist
- [x] [Corresponding issue has been opened](https://tech-matters.atlassian.net/browse/CHI-2022)
- [x] New tests added

### Verification steps
- Start development server - this branch for HRM (`npm run ssm:local & npm start`), master for Serverless (`npm run start:local`) and Flex (`npm run dev:local:serverless`).
- Start a new chat contact and submit it.
- Wait for the transcript to be exported.
- Confirm  that in all places where a contact can be accessed, the transcript is too (search contacts, search cases, cases list).
- Confirm that in the DB the records are as expected:
  - `Contact` -> `rawJson` does not contains the `conversationMedia` anymore.
  - A new S3 type record exists in `ConversationMedia` for the above contact, with the appropriate information (bucket and location).
  - The `ContactJob` for the transcript export is appropriately marked as complete.

### CODE REVIEW

File: hrm-domain/hrm-service/src/conversation-media/conversation-media-data-access.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+type TwilioStoredMedia = ConversationMediaCommons & NewTwilioStoredMedia;
+
+type NewS3StoredTranscript = {
+  storeType: 'S3';
+  storeTypeSpecificData: {
+    type: S3ContactMediaType.TRANSCRIPT;
+    location?: string;
+    url?: string;
+  };
+};
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/conversation-media/conversation-media-data-access.ts:61]: Check if this is duplicate of transcript job payload. If so, unify.

File: hrm-domain/hrm-service/src/contact/contact.ts [hunk 2]
```diff
@@ -117,11 +158,16 @@ const permissionsBasedTransformations: PermissionsBasedTransformation[] = [
...
+    const permissionsBasedTransformed = permissionsBasedTransformations.reduce(
       (transformed, { action, transformation }) =>
         !can(user, action, contact) ? transformation(transformed) : transformed,
       contact,
     );
+
+    // TODO: Remove once all Flex clients are using new ConversationMedia model
+    const transformed = addLegacyConversationMedia(permissionsBasedTransformed);
+
+    return transformed;
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/contact/contact.ts:170]: Emphasize this has to be last step for permission reasons, and make sure we are filtering legacy stuff from blob.

**Review by robert-bo-davis — APPROVED** [review]
LGTM. I tested locally and your code worked, but [mine was broken](https://github.com/techmatters/flex-plugins/pull/1594/files). 😿

File: hrm-domain/hrm-service/src/conversation-media/conversation-media-data-access.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  id: ConversationMedia['id'],
+  storeTypeSpecificData: ConversationMedia['storeTypeSpecificData'],
+): Promise<void> =>
+  db.task(async connection =>
+    connection.none(updateSpecificDataByIdSql, {
+      accountSid,
+      id,
+      storeTypeSpecificData,
+    }),
+  );
```
Comment (GPaoloni) [inline hrm-domain/hrm-service/src/conversation-media/conversation-media-data-access.ts:131]: Explicit update of timestamp is required?

**Comment by robert-bo-davis** [comment]
Migration and updates LGTM

**Pull request merged**